### PR TITLE
Create New UI Test for MainMenuActivity Toolbar

### PR DESCRIPTION
**Pull request by Stryker101**
# This is a pull request raised for issue https://github.com/odk-x/tool-suite-X/issues/483

## **Reason for the PR
The  toolbar of the Survey App was recently revised. presently it interacts with the Services App through intents depending on user interactions. This PR add a test to verify that these interactions are working as they were intended.

## **How the Test Works
Please note that the Services App is required to be installed on the testing device or emulator to be used.
Since the list of forms is an asynchronous task, i added MockFormData and MockFormListLoader to simulate this process 

these makes sure a list of form is available during the setUp() of the test

the test is also configured with the required permissions needed to run the app

## **Functionality Tested
This Test verifies the following
-The correct intent is sent when the sync icon is clicked
-The correct intent is sent when the settings icon is clicked
-The behaviour when sorted by Name
-The behaviour when sorted by TableId

Screenshot 2024-03-31 111138

### CODE REVIEW

**Review by wbrunette — CHANGES_REQUESTED** [review]
Overall, good job.

To be able to get it to run successfully, you should add to survey's gradle file (similar to Services). There was a problem with animations.   

  testOptions {
        animationsDisabled = true
        unitTests {
            includeAndroidResources = true
        }
    }

File: survey_app/build.gradle [hunk 1]
```diff
@@ -126,13 +132,25 @@ dependencies {
...
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
+    androidTestImplementation 'androidx.test.espresso:espresso-intents:3.5.1'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.5'
+    androidTestImplementation 'org.mockito:mockito-core:2.19.0'
+    androidTestImplementation 'com.google.truth:truth:1.1.3'
+    androidTestImplementation 'androidx.test:core:1.5.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-contrib:3.5.1'
+
+    testImplementation 'junit:junit:4.13.2'
+    testImplementation 'org.mockito:mockito-core:2.19.0'
```
Comment (wbrunette) [inline survey_app/build.gradle]: I missed this before, why are you adding the dependencies to "testImplementation" if you only need them when you run androidTest (note: tests in "androidTest" directory use androidTestImplementation dependencies, tests in "test" directory uses testImplementation dependencies). You did not add tests to the test directory so not sure why needed

**Review by wbrunette — CHANGES_REQUESTED** [review]
normally mock objects used for testing reside in the test directories they are used in so they are not included in the release artifact. Unnecessary classes make the release bigger and slower.

File: survey_app/build.gradle (hunk 1 above)
Comment (Stryker101) [reply survey_app/build.gradle]: Thanks for the feedback. I'll take those out.

**Review by wbrunette — CHANGES_REQUESTED** [review]
Tests are not working, check logs.

org.opendatakit.survey.activities.MainMenuActivityTest > testSortByName[avd(AVD) - 11] FAILED 
        java.lang.ClassCastException: org.opendatakit.survey.fragments.InitializationFragment cannot be cast to org.opendatakit.survey.fragments.FormChooserListFragment
        at org.opendatakit.survey.activities.MainMenuActivityTest.lambda$setUp$0$org-opendatakit-survey-activities-MainMenuActivityTest(MainMenuActivityTest.java:65)

org.opendatakit.survey.activities.MainMenuActivityTest > testSyncToolbarItem_whenCLickedShouldSendCorrectIntent[avd(AVD) - 11] FAILED 
        java.lang.ClassCastException: org.opendatakit.survey.fragments.InitializationFragment cannot be cast to org.opendatakit.survey.fragments.FormChooserListFragment
        at org.opendatakit.survey.activities.MainMenuActivityTest.lambda$setUp$0$org-opendatakit-survey-activities-MainMenuActivityTest(MainMenuActivityTest.java:65)

org.opendatakit.survey.activities.MainMenuActivityTest > testSettingsToolbarItem_whenCLickedShouldSendCorrectIntent[avd(AVD) - 11] FAILED 
        java.lang.ClassCastException: org.opendatakit.survey.fragments.InitializationFragment cannot be cast to org.opendatakit.survey.fragments.FormChooserListFragment
        at org.opendatakit.survey.activities.MainMenuActivityTest.lambda$setUp$0$org-opendatakit-survey-activities-MainMenuActivityTest(MainMenuActivityTest.java:65)

org.opendatakit.survey.activities.MainMenuActivityTest > testSortByTableId[avd(AVD) - 11] FAILED 
        java.lang.ClassCastException: org.opendatakit.survey.fragments.InitializationFragment cannot be cast to org.opendatakit.survey.fragments.FormChooserListFragment
        at org.opendatakit.survey.activities.MainMenuActivityTest.lambda$setUp$0$org-opendatakit-survey-activities-MainMenuActivityTest(MainMenuActivityTest.java:65)

**Comment by Stryker101** [comment]
@wbrunette i added @Ignore to one of the test cases (testSettingsToolbarItem_whenCLickedShouldSendCorrectIntent()) because clicking that button launches the services app, hence that particular test needs services to run, and the services pipeline i'snt ready yet.

**Review by wbrunette — APPROVED** [review]

**Pull request merged**